src/onnx/onnx_parser.cpp

```diff
@@ -375,8 +375,11 @@ void onnx_parser::parse_graph(module* mod, const onnx::GraphProto& graph)
     // add the return instuction
     mod->add_return(output_ins);
 
-    // remove instructions added in this mod
-    erase_if(instructions, [&](auto&& p) { return mod->has_instruction(p.second); });
+    if(not inlining)
```

Comment:
we can also not just add `add_return` here if we are inlining a subgraph.

Reply:
But to do this, we need to have `parse_graph` return the vector of returned instructions.

Reply:
Done. Just gonna let this run till overnight. Used (void) to signal ignored returns in our code since we want to support earlier versions of modern C++ and only since C+17  [[maybe_unused]] was added.